.github/workflows/prover.yml

```
@@ -0,0 +1,41 @@
+name: Run prover guest code checks
+
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          toolchain: nightly-2024-07-27
+      
+      # Installs mold (modern ld), a drop-in replacement for lld.
+      # Under the hood, the following actions symlinks mold binary onto lld,
+      # so everything is linked faster (hopefully).
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
+
+      - name: Set up Clang
+        uses: egor-tensin/setup-clang@v1
+        with:
+          version: latest
+          platform: x64
+
+      - name: Install SP1 Toolchain
+        run: |
+          curl -L https://sp1.succinct.xyz | bash && ~/.sp1/bin/sp1up
+          export PATH=~/.sp1/bin:$PATH
+          cargo prove --version
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build guest code
+        run: cd provers/sp1/ && RUSTFLAGS="-C link-arg=-fuse-ld=lld" cargo build --profile prover-ci -F "prover"
```

```suggestion
        run: cargo build --profile prover-ci -F "prover"
        working-directory: provers/sp1
        env:
          RUSTFLAGS: "-C link-arg=-fuse-ld=lld"
```

see https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory

and https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsenv

---

Hm, nice, thanks!